### ISSUE TITLE
Django 3.2 breaks django-multisite: The SITE_ID setting must be an integer

### ISSUE THREAD

**Issue by digodude** [ecometrica/django-multisite#76]
After upgrading to Django 3.2 and using django-multisite 1.9.0 the development django runserver doesn't pass a new test:

ERRORS:
?: (sites.E101) The SITE_ID setting must be an integer

**Comment by erimeilis** [comment]
Yep, and if you use SITE_ID = int(SiteID(default=1)) instead, than you'll get:

env/lib/python3.9/site-packages/multisite/middleware.py", line 46, in __init__
    raise TypeError('Invalid type for settings.SITE_ID: %s' %
TypeError: Invalid type for settings.SITE_ID: int

**Comment by pfouque** [comment]
Hi, i didn't tried yet but there is a fix for that issue in parent project : https://github.com/shestera/django-multisite/commit/1ec217d075f05e9110b933354035dbee136e0674

**Comment by erimeilis** [comment]
> Hi, i didn't tried yet but there is a fix for that issue in parent project : [shestera@1ec217d](https://github.com/shestera/django-multisite/commit/1ec217d075f05e9110b933354035dbee136e0674)

Hello! It seems to eliminate problem with SITE_ID, but another one comes, - 404 for every site.

**Comment by mellertson** [comment]
I undertstand using `SILENCED_SYSTEM_CHECKS` will quiet the error message.  But, I'm just curious if another solution has been found for this issue?

**Comment by mellertson** [comment]
I found that doing `SITE_ID = int(os.environ.get('SITE_ID', 1))` fixed the issue with Django version 4.2.  Hope that helps someone.
